presto-clickhouse/src/main/java/com/facebook/presto/plugin/clickhouse/StandardReadMappings.java

```diff
@@ -140,7 +141,8 @@ public static ReadMapping timestampReadMapping()
     {
         return longReadMapping(TIMESTAMP, (resultSet, columnIndex) -> {
             Timestamp timestamp = resultSet.getTimestamp(columnIndex);
-            return timestamp.getTime();
+            // getTimestamp loses the milliseconds, but we can get them from the getString
```

Comment:
What do you mean here? 


 /**
     * Returns the number of milliseconds since January 1, 1970, 00:00:00 GMT
     * represented by this {@code Timestamp} object.
     *
     * @return  the number of milliseconds since January 1, 1970, 00:00:00 GMT
     *          represented by this date.
     * @see #setTime
     */
    public long getTime() {
        long time = super.getTime();
        return (time + (nanos / 1000000));
    }

Reply:
do you mean for precision? can we use nanos then?

```
/**
     * Gets this {@code Timestamp} object's {@code nanos} value.
     *
     * @return this {@code Timestamp} object's fractional seconds component
     * @see #setNanos
     */
    public int getNanos() {
        return nanos;
    }
```

Reply:
That if there's a timestamp string like 2025-01-08 06:34:56.179, the .179 is lost (3 digits have a precision of milliseconds)

Reply:
even if we use nanos?

Reply:
Yeah, Nanos from the timestamp is always 0